sansio_lsp_client/structs.py

```diff
@@ -51,11 +54,15 @@ class VersionedTextDocumentIdentifier(TextDocumentIdentifier):
     version: t.Optional[int]
 
 
+# Sorting tip:  sorted(positions, key=(lambda p: (p.line, p.character)))
 class Position(BaseModel):
     # NB: These are both zero-based.
     line: int
     character: int
 
+    def as_tuple(self) -> t.Tuple[int, int]:
+        return (self.line, self.character)
```

Comment:
@halfbrained My `as_tuple()` method replaces `__lt__` in this class and `pos()` in `class DocumentSymbol`. What do you think?

Reply:
After making this review comment, I changed the lambda to `lambda p: p.as_tuple()`, which is even simpler.

Reply:
Is there a benefit to this change? :)

Reply:
I think `Position.__lt__()` and `DocumentSymbol.pos()` were both weird:
- If there is `__lt__`, why are `__gt__`, `__le__` and `__ge__` missing? It's because `__lt__` is what `sort()` and `sorted()` use, but it's not obvious when looking at the code. We could add a comment about it, but `key=(lambda p: p.as_tuple())` is very obvious as is.
- If you see `something.pos()` in the middle of longer code, you wouldn't expect it to take just the **start** of a range and ignore the end, which is what that method did. On the other hand, if you see `something.selectionRange.start.as_tuple()`, you immediately know it uses just the start position and the end is being ignored.